lib/node-utils.ts

```diff
@@ -180,6 +266,86 @@ export function getVariableReferences(
   );
 }
 
+interface InnermostFunctionScope extends TSESLintScope.FunctionScope {
+  block:
+    | TSESTree.FunctionDeclaration
+    | TSESTree.FunctionExpression
+    | TSESTree.ArrowFunctionExpression;
+}
+
+export function getInnermostFunctionScope(
+  context: RuleContext<string, []>,
+  asyncQueryNode: TSESTree.Identifier
+): InnermostFunctionScope | null {
+  const innermostScope = ASTUtils.getInnermostScope(
+    context.getScope(),
+    asyncQueryNode
+  );
+
+  if (
+    innermostScope?.type === 'function' &&
+    ASTUtils.isFunction(innermostScope.block)
+  ) {
+    return (innermostScope as unknown) as InnermostFunctionScope;
+  }
+
+  return null;
+}
+
+export function getFunctionReturnStatementNode(
+  functionNode:
+    | TSESTree.FunctionDeclaration
+    | TSESTree.FunctionExpression
+    | TSESTree.ArrowFunctionExpression
+): TSESTree.Node | null {
+  if (isBlockStatement(functionNode.body)) {
+    // regular function or arrow function with block
+    const returnStatementNode = functionNode.body.body.find((statement) =>
+      isReturnStatement(statement)
+    ) as TSESTree.ReturnStatement | null;
```

Comment:
if not found, `find` will return `undefined` instead of `null`

```suggestion
    const returnStatementNode = functionNode.body.body.find((statement) =>
      isReturnStatement(statement)
    ) as TSESTree.ReturnStatement | undefined;
```

Reply:
Nice one. Done.